src/library/scala/runtime/LazyRef.scala

```diff
@@ -0,0 +1,52 @@
+package scala.runtime
+
+/* Classes used as holders for local lazy vals */
+
+class LazyRef[T] {
+  var value: T = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyBoolean {
+  var value: Boolean = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyByte {
+  var value: Byte = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyChar {
+  var value: Char = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyShort {
+  var value: Short = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyInt {
+  var value: Int = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyLong {
+  var value: Long = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyFloat {
+  var value: Float = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyDouble {
+  var value: Double = _
+  @volatile var initialized: Boolean = false
+}
+
+class LazyUnit {
+  @volatile var initialized: Boolean = false
+}
```

Comment:
Compare these with `VolatileXxxRef`, e.g.

```java
public class VolatileLongRef implements java.io.Serializable {
    private static final long serialVersionUID = -3567869820105829499L;

    volatile public long elem;
    public VolatileLongRef(long elem) { this.elem = elem; }
    public String toString() { return java.lang.Long.toString(elem); }

    public static VolatileLongRef create(long e) { return new VolatileLongRef(e); }
    public static VolatileLongRef zero() { return new VolatileLongRef(0); }
}
```

Differences to discuss:
  - implementation in Java vs Scala
  - Serializability
  - toString